bluepyemodel/tasks/emodel_creation/optimisation.py

```diff
@@ -155,8 +155,21 @@ def output(self):
         targets = []
         mechanisms_directory = self.access_point.get_mechanisms_directory()
 
+        architectures = ["x86_64", "i686", "powerpc", "umac"]
+
+        # Detect the first existing architecture
+        arch_dir = None
+        for arch in architectures:
+            dir_to_check = mechanisms_directory.parents[0] / arch
+            if dir_to_check.exists() and dir_to_check.is_dir():
+                arch_dir = dir_to_check
+                break
+
+        if arch_dir is None:
+            raise ValueError("No valid architecture directory found")
```

Comment:
When we begin the workflow, there is a large chance mechanisms are not compiled, and so we will just raise an error in the middle of the workflow, which is not good.
I guess in this case you could just set arch_dir to whatever architecture, since the targets will not be present anyway

Reply:
You're absolutely right, I will create a new PR

Reply:
This PR should take care of that: https://github.com/BlueBrain/BluePyEModel/pull/80